### PR TITLE
Added GUI alert on mailer failure.

### LINKED ISSUE #219: No way to notify users via UI in case of mail failure
When sending mail fails, [only a small message is printed via NSLog](https://github.com/Lickability/PinpointKit/blob/c3bbb00f8cf605360abfc93b0147e1eb626dca91/PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift#L89). There seems to be no way to configure PinpointKit to react to this situation in the GUI.

I'd like instead to pop up an alert to notify the user via the UI, and dismiss the PinpointKit view controller, but as best as I can tell the current design doesn't allow me to do so either via delegates (`PinpointKitDelegate` has no "mail failed" function) or overrides (as I cannot set the `SenderDelegate` to a new class and still get access to PinpointKit's private `displayingViewController` within that class, which I would need to dismiss the view on successful sending like the default does). I also need to do this in the case where the view is brought up via shaking, so I can't assume I just have a reference to the presenting VC.

### PR DESCRIPTION
**Pull request by sxmichaels**
Closes #219.

## What It Does
Previously, if the user clicked `Send` and the `MailSender` experienced a failure (e.g. when there are no mail accounts set up), nothing would happen in the GUI. There was also no method in `PinpointKitDelegate` to correspond to `didFailToSend` in `SenderDelegate`.

This commit adds an alert popup to `PinpointKit` to solve the first problem, and a `didFailToSend` method to `PinpointKitDelegate` for the second. The latter takes the original `Error` as a parameter in case developers want to use it present more specific messages.

simulator

## How to Test
Ensure that the mailer will encounter an error (the easiest way is to run the example app on a simulator with no email accounts set up) and click `Send`.

## Notes
None in particular.

### CODE REVIEW

File: PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift [hunk 1]
```diff
@@ -86,7 +86,18 @@ extension PinpointKit: SenderDelegate {
     public func sender(_ sender: Sender, didFailToSend feedback: Feedback?, error: Error) {
         if case MailSender.Error.mailCanceled = error { return }
         
-        NSLog("An error occurred sending mail: \(error)")
+        let alert = UIAlertController(
```
Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift]: Can you move this UIAlertController and presentation to the default protocol extension on ` func pinpointKit(_ pinpointKit: PinpointKit, didFailToSend feedback: Feedback, error: Error) {}`?

File: PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift [hunk 2]
```diff
@@ -86,7 +86,18 @@ extension PinpointKit: SenderDelegate {
...
+            title: "Mailer error",
+            message: "Unable to prepare an email to send. Ensure that you have at least one email account set up.",
+            preferredStyle: UIAlertControllerStyle.alert)
```
Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift]: Please use `.alert` without the  `UIAlertControllerStyle` preceding it.

File: PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift [hunk 3]
```diff
@@ -108,11 +119,21 @@ public protocol PinpointKitDelegate: class {
      - parameter feedback:      The feedback that’s just been sent.
      */
     func pinpointKit(_ pinpointKit: PinpointKit, didSend feedback: Feedback)
+    
+    /**
+     Notifies the delegate that PinpointKit has failed to send user feedback.
```
Comment (mattbischoff) [inline PinpointKit/PinpointKit/Sources/Core/PinpointKit.swift]: Thanks for adding docs!

**Comment by sxmichaels** [comment]
OK, I made the above changes.

**Pull request merged**